File: EVA2_22_WHILE_GAME.py
import random as rm

iRandom = rm.randint(1, 10)

userValue = 0
intentos = 0
i = 0

print('Adivina el numero del 1 al 10')
print(iRandom)

while userValue != iRandom:
    userValue = int(input('Introduce tu numero para adivinar: '))
    if userValue > iRandom:
        print('Te has pasado')
    elif userValue < iRandom:
        print('Te falta poco')

    while i != 3:
        print(" ")
        i = i + 1
    intentos = intentos + 1

print(f'Has ganado, adivinaste el numero en {intentos} intentos.')

File: EVA2_22_WHILE_GAME/src/eva2_22_while_game/EVA2_22_WHILE_GAME.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package eva2_22_while_game;

import java.util.Scanner;

/**
 *
 * @author riguidix
 */
public class EVA2_22_WHILE_GAME {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        // TODO code application logic here
        // GENERAR EL NUMERO 1 Y 10
        // NO DEJEN TERMINAR AL USUARIO HASTA QUE ADIVINE
        Scanner input = new Scanner(System.in);
        
        int iRandom = (int) (Math.random() * 10);
        
        if (iRandom == 0) {
            iRandom = (int) (Math.random() * 1);
        }
        
        int userValue = 0;
        int intentos = 0;
        
        
        System.out.println("Adivina el numero del 1 al 10");
        System.out.println(iRandom);
        while (userValue != iRandom) {            
            System.out.println("Introduce el numero para adivinar");
            userValue = input.nextInt();
            if (userValue > iRandom) {
                System.out.println("Te has pasado");
            } else if (userValue < iRandom){
                System.out.println("Te falta poco");
            }
            
            for (int i = 0; i < 3; i++) {
                System.out.println(" ");
            }
            intentos++;
        }
        System.out.println("Has ganado, adivinaste el numero en " + intentos + " intentos.");
    }
    
}
